### PR TITLE
Redesign of MarketList

### PR DESCRIPTION
**Pull request by magnustviberg**
# Why?

The market list design has been updated as part of the new front page

# What?

New design for both MarketsGridView and CompactMarketsView. AdRecommendationCells has now got a .clear background since the entire front page background color has changed. 

# Version Change

Minor

# UI Changes

| Before | After |
| --- | --- |
| Simulator Screen Shot - iPhone 12 Pro - 2021-09-15 at 13 32 06 | Simulator Screen Shot - iPhone 12 Pro - 2021-09-15 at 13 27 36 |

**Gif**
Simulator Screen Recording - iPhone 12 Pro - 2021-09-15 at 13 27 54

### CODE REVIEW

**Review by bstien — APPROVED** [review]
Great job!🎉 The redesign looks awesome😄

File: FinniversKit/Sources/Fullscreen/FrontPage/FrontPageView.swift [hunk 1]
```diff
@@ -8,11 +8,16 @@ public protocol FrontPageViewModel {
...
 }
 
-public protocol FrontPageViewDelegate: AnyObject {
+public protocol FrontPageViewDelegate: MarketsViewDelegate, MarketsViewDataSource, AdRecommendationsGridViewDelegate {
     func frontPageViewDidSelectRetryButton(_ frontPageView: FrontPageView)
 }
 
+protocol CompactMarketsPresenter {
+    func scrollViewDidScroll(_ scrollView: UIScrollView)
+}
```
[the remark this answers is not in the document]
Comment (magnustviberg) [reply FinniversKit/Sources/Fullscreen/FrontPage/FrontPageView.swift:17]: No its not suppose to be there 😆

File: FinniversKit/Sources/Fullscreen/FrontPage/FrontPageView.swift [hunk 2]
```diff
@@ -28,17 +33,40 @@ public final class FrontPageView: UIView, BasicFrontPageView {
...
+    
+    private enum CompactMarketsViewVisibilityStatus {
+        case hidden
+        case displaying(progress: CGFloat)
+        case displayed
+    }
 
-    private weak var delegate: FrontPageViewDelegate?
-
+    private let delegate: FrontPageViewDelegate
```
[the remark this answers is not in the document]
Comment (magnustviberg) [reply FinniversKit/Sources/Fullscreen/FrontPage/FrontPageView.swift]: Done 👍 https://github.com/finn-no/FinniversKit/pull/1059/commits/5c9155799c6fff6f266dfe95190596d16ed788d6

File: FinniversKit/Sources/Recycling/GridViews/Markets/GridView/Cell/MarketsGridViewCell.swift [hunk 3]
```diff
@@ -6,7 +6,43 @@ import UIKit
 
 class MarketsGridViewCell: UICollectionViewCell {
     // MARK: - Internal properties
-
+    
+    private let cornerRadius: CGFloat = 16
+    
+    private lazy var sharpShadowView: UIView = {
+        let view = UIView()
+        view.translatesAutoresizingMaskIntoConstraints = false
```
[the remark this answers is not in the document]
Comment (magnustviberg) [reply FinniversKit/Sources/Recycling/GridViews/Markets/GridView/Cell/MarketsGridViewCell.swift]: Done 👍 https://github.com/finn-no/FinniversKit/pull/1059/commits/4026ca5cd4bf12df12208a01385c7f4f5539da57

File: FinniversKit/Sources/Recycling/GridViews/Markets/CompactView/CompactMarketsView.swift [hunk 4]
```diff
@@ -131,3 +156,24 @@ extension CompactMarketsView: UICollectionViewDelegate {
...
+private extension CompactMarketsView {
+    func styleShadowAfterLayout() {
+        smoothShadowView.layer.shadowPath = UIBezierPath(rect: CGRect(x: 0,
+                                                                        y: smoothShadowView.bounds.maxY - smoothShadowView.layer.shadowRadius,
+                                                                        width: smoothShadowView.bounds.width,
+                                                                        height: smoothShadowView.layer.shadowRadius)).cgPath
+        sharpShadowView.layer.shadowPath = UIBezierPath(rect: CGRect(x: 0,
+                                                                        y: sharpShadowView.bounds.maxY - sharpShadowView.layer.shadowRadius,
+                                                                        width: sharpShadowView.bounds.width,
+                                                                        height: sharpShadowView.layer.shadowRadius)).cgPath
```
[the remark this answers is not in the document]
Comment (bstien) [reply FinniversKit/Sources/Recycling/GridViews/Markets/CompactView/CompactMarketsView.swift]: I'm sorry, I see I misplaced the `.cgPath` 😿 It should of course be called from the `UIBezierPath`

**Pull request merged**